### ISSUE TITLE
Map not rendering due to "Parameter 'lon' must be a number."

### ISSUE THREAD

**Issue by chewitt** [FriendsOfFlarum/geoip#35]
## Bug Report

**Current Behavior**
Flarum was clean-installed in late November and installed the plugin. I can see GeoIP data is retrieved fine using multiple services but the map data never renders and the following error is shown:

```
Object { status: 400, options: {…}, xhr: XMLHttpRequest, responseText: `{"error":{"code":400,"message":"Parameter 'lon' must be a number."}}`, response: {…}, alert: {…} }
​alert: Object { type: "error", content: (1) […], controls: false }
​options: Object { background: true, url: "https://nominatim.openstreetmap.org/reverse", method: "GET", … }
​response: Object { error: {…} }
​responseText: `{"error":{"code":400,"message":"Parameter 'lon' must be a number."}}`
​status: 400
​xhr: XMLHttpRequest { readyState: 4, timeout: 0, withCredentials: false, … }
​<prototype>: Object { … }
```
I've had several users/staff validate the error is geo-consistent (broken everywhere).

**Steps to Reproduce**
1. Install plugin
2. Set provider
3. Check session data for self (or any user..)
4. Notice the map never renders

**Expected Behavior**
Looking at a pretty map :)

**Screenshots**
<img width="928" alt="image" src="https://github.com/FriendsOfFlarum/geoip/assets/251794/402f9565-babd-4895-818a-ab8b9fe90f2a">

**Environment**
- Flarum version: 1.8.5
- Extension version: 1.4.1
- Website URL: xxxx
- Webserver: nginx
- Hosting environment: vps
- PHP version: 8.1
- Browser: Firefox 122 on macOS

```
Flarum core: 1.8.5
PHP version: 8.1.2-1ubuntu2.14
MySQL version: 8.0.36-0ubuntu0.22.04.1
Loaded extensions: Core, date, libxml, openssl, pcre, zlib, filter, hash, json, pcntl, Reflection, SPL, session, standard, sodium, mysqlnd, PDO, xml, calendar, ctype, curl, dom, mbstring, FFI, fileinfo, ftp, gd, gettext, iconv, exif, mysqli, pdo_mysql, Phar, posix, readline, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, xmlreader, xmlwriter, xsl, zip, Zend OPcache
+-------------------------------------+---------+--------+
| Flarum Extensions                   |         |        |
+-------------------------------------+---------+--------+
| ID                                  | Version | Commit |
+-------------------------------------+---------+--------+
| flarum-flags                        | v1.8.0  |        |
| flarum-suspend                      | v1.8.1  |        |
| flarum-tags                         | v1.8.0  |        |
| flarum-approval                     | v1.8.1  |        |
| fof-user-directory                  | 1.3.3   |        |
| nearata-copy-code-to-clipboard      | v2.2.0  |        |
| matteociaroni-public-suspensions    | v1.0.0  |        |
| ianm-twofactor                      | 1.0.11  |        |
| ianm-syndication                    | 1.3.4   |        |
| hikarilan-passkey-login             | 0.2     |        |
| fof-user-bio                        | 1.3.2   |        |
| fof-upload                          | 1.5.4   |        |
| fof-terms                           | 1.3.0   |        |
| fof-subscribed                      | 1.1.4   |        |
| fof-split                           | 1.1.1   |        |
| fof-recaptcha                       | 1.3.2   |        |
| fof-reactions                       | 1.4.0   |        |
| fof-pages                           | 1.0.8   |        |
| fof-open-collective                 | 1.1.1   |        |
| fof-nightmode                       | 1.5.3   |        |
| fof-merge-discussions               | 1.4.1   |        |
| fof-links                           | 1.2.2   |        |
| fof-github-autolink                 | 1.1.4   |        |
| fof-geoip                           | 1.4.1   |        |
| fof-formatting                      | 1.0.3   |        |
| fof-drafts                          | 1.2.10  |        |
| fof-default-group                   | 1.1.2   |        |
| fof-cookie-consent                  | 1.1.2   |        |
| fof-best-answer                     | 1.4.1   |        |
| fof-anti-spam                       | 1.1.2   |        |
| flarum-subscriptions                | v1.8.0  |        |
| flarum-sticky                       | v1.8.0  |        |
| flarum-statistics                   | v1.8.0  |        |
| flarum-nicknames                    | v1.8.0  |        |
| flarum-mentions                     | v1.8.3  |        |
| flarum-markdown                     | v1.8.0  |        |
| flarum-lock                         | v1.8.0  |        |
| flarum-lang-english                 | v1.8.0  |        |
| flarum-emoji                        | v1.8.0  |        |
| flarum-bbcode                       | v1.8.0  |        |
| extiverse-mercury                   | 0.2.0   |        |
| clarkwinkelmann-first-post-approval | 1.0.1   |        |
| clarkwinkelmann-circle-groups       | 1.0.1   |        |
| blomstra-fontawesome                | 0.1.5   |        |
| blomstra-database-queue             | 1.1.0   |        |
| antoinefr-online                    | v1.0.1  |        |
| acpl-mobile-tab                     | 1.4.2   |        |
+-------------------------------------+---------+--------+
Base URL: xxxx
Installation path: /var/www/xxxx
Queue driver: database
Session driver: file
Scheduler status: Active
Mail driver: smtp
Debug mode: off
```

**Possible solution(s)**

**Additional Context**
N/A

**Comment by imorland** [comment]
It would appear that there is no value for either zip code or lat/long 🤔 

Can you please confirm
- Which provider you are using
- If the zip code is displayed
- If possible, confirm if the db row for this IP contains the zip and/or lat/long

**Comment by chewitt** [comment]
The forum is low-traffic and only went live on Monday so there's not much data in the DB and the config was changed at some point. Looking at the DB shows that entries sourced from ip-api are complete with lat/lon/zip data. All other entries have been sourced from IPlocation and have basic data on `country_code` and `isp` but otherwise have null values.

NB: I've switched the forum back to IP-Api for now, but happy to validate any patches/changes.
